futurex_openedx_extensions/helpers/tenants.py

```diff
@@ -369,3 +373,139 @@ def create_new_tenant_config(sub_domain: str, platform_name: str) -> TenantConfi
+        return get_nested_value(config, path)
+
+    if draft_only:
+        return get_nested_value(config.get('config_draft', {}), path)
+
+    return get_nested_value(config.get('config_draft', {}), path) or get_nested_value(config, path)
+
+
+def get_draft_tenant_config(tenant_id: int) -> dict:
+    """
+    This function fetches the configuration for the specified tenant and returns the
+    fields that differ between the published and draft configurations.
+
+    :param tenant_id: The ID of the tenant whose draft configuration are to be retrieved.
+    :type tenant_id: int
+    :return: A dictionary containing updated fields with published and draft values.
+    :rtype: dict
+    """
+    config = get_all_tenants_info()['config'].get(tenant_id, {})
+    updated_fields = {}
+    for field_config in ConfigAccessControl.objects.all():
+        draft_value = get_tenant_config_value(config, field_config.path, published_only=False, draft_only=True)
+        published_value = get_tenant_config_value(config, field_config.path, published_only=True, draft_only=False)
+        if draft_value and published_value:
+            updated_fields[field_config.key_name] = {
+                'published_value': published_value,
+                'draft_value': draft_value,
+            }
+    return updated_fields
+
+
+def update_draft_tenant_config(
+    tenant_id: int, key_access_info: ConfigAccessControl, current_value: str, new_value: str, reset: bool = False
+) -> None:
+    """
+    Updates the draft configuration for the specified tenant with the new value.
+    If reset is True, it resets the field to its published value or removes it.
+
+    :param tenant_config: The configuration for the tenant, containing both published and draft data.
+    :type tenant_config: dict
+    :param key_access_info: A dictionary or path representing the nested key in the configuration.
+    :type key_access_info: dict or list of strings
+    :param new_value: The new value to update the field with.
+    :type new_value: str
+    :param reset: A boolean indicating whether to reset the field to its published value. Defaults to False.
+    :type reset: bool
+    :return: A dictionary with the updated fields and their new values.
+    :rtype: dict
+    """
+    config = get_all_tenants_info()['config'].get(tenant_id)
+    if not config:
+        raise FXCodedException(
+            code=FXExceptionCodes.INVALID_INPUT,
+            message=f'Unable to find config for tenant: {tenant_id} or maybe it is empty.'
+        )
+
+    stored_value = get_tenant_config_value(config, key_access_info.path)
+    if not stored_value:
+        raise FXCodedException(
+            code=FXExceptionCodes.INVALID_INPUT,
+            message='Invalid path for key: ({key_access_info.key}), path: ({key_access_info.path}) '
+            'as it does not exist in tenant config for tenant or maybe its value is empty: {tenant_id}.'
+        )
+    if stored_value != current_value:
```

Comment:
we can let MySQL handle the comparison while updating the key. This is good, but it'll not guarantee error detection for parallel write commands